### PR TITLE
Fixes CNS rebooter.

### LINKED ISSUE #43374: CNS rebooter not functional
CNS rebooter should cap stuns to 4 seconds, but it appears to do nothing at the moment.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Fixes CNS rebooter. Fixes #43374.

## Why It's Good For The Game

Fixes CNS rebooter

## Changelog
:cl:
fix: The CNS rebooter now purges stuns after 4 seconds of not being stunned, instead of doing nothing.
/:cl:

### CODE REVIEW

**Comment by XDTM** [comment]
It's supposed to reduce all stuns to a maximum amount, not completely negate them

**Comment by nemvar** [comment]
Well, and you see how well that worked. This is the same in the vast majority of cases, except that additional stun reduction doesn't effect it as much. There isn't that big of a difference between capping stuns amount at 4 seconds and purging all stuns after 4 second of not being stunned.

I had four choices:
1. Shitcode physiology to allow for maximum stun amount.
2. Make shitcode that sends a signal AFTER Stun() has done its thing.
3. Refactor status effects.
4. This.

Not to talk about how this much better fits the flavour of a rebooter. If this would cap stuns, it should be called limiter or something.

**Comment by XDTM** [comment]
My bad, i actually misread the code and didn't see that the effect was under a timer. 
This actually seems better than the original implementation, although i'd also add some kind of feedback to the user or maybe to all viewers that it's about to trigger, or it's going to feel very random.

File: code/modules/surgery/organs/augments_internal.dm [hunk 1]
```diff
@@ -117,19 +116,16 @@
...
-		owner.SetStun(stun_cap_amount)
-	if(owner.AmountKnockdown() > stun_cap_amount)
-		owner.SetKnockdown(stun_cap_amount)
-	if(owner.AmountImmobilized() > stun_cap_amount)
-		owner.SetImmobilized(stun_cap_amount)
-	if(owner.AmountParalyzed() > stun_cap_amount)
-		owner.SetParalyzed(stun_cap_amount)
-	working = FALSE
+/obj/item/organ/cyberimp/brain/anti_stun/proc/on_signal(datum/source, amount)
+	if(!broken_cyber_organ && amount > 0)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/organs/augments_internal.dm]: what does the amount var do and does it change?

Comment (nemvar) [reply code/modules/surgery/organs/augments_internal.dm]: Stun reductions (for example from chems) send us a negative amount which would reset the timer if this check weren't here.

Basically, nicotine breaks the implant without this.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/organs/augments_internal.dm]: lovely

**Pull request merged**